app/controllers/users_controller.rb

```diff
@@ -1,9 +1,12 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by id: params[:id]
-    return if @user
-    flash[:warning] = t ".nouser"
-    redirect_to root_path
+  before_action :load_user, except: %i(index correct_user update destroy create)
+  before_action :logged_in_user, only: %i(index edit update)
+  before_action :correct_user, only: %i(edit update)
+  before_action :verify_admin, only: :destroy
+
+  def show; end
+  def index
+    @users = User.select(:id, :name, :email).order(created_at: :DESC).paginate(page: params[:page], per_page: Settings.perpage)
```

Comment:
1. Maximum 80 characters per line 
2. .select(:id, :name, :email).order(created_at: :DESC)  -> for this create a scope in the model, then call this